library/src/main/java/com/davidstemmer/screenplay/flow/Screenplay.java

```diff
@@ -215,9 +238,20 @@ private void attachToParent(Stage stage, View view) {
         stage.getContainer().addView(view);
     }
 
-    private void removeFromParent(View view) {
-        ViewGroup parent = (ViewGroup) view.getParent();
-        parent.removeView(view);
+    private void removeFromParent(final View view) {
+
+        // If we don't include the post() call, the following error occurs:
+        // Attempt to read from field 'int android.view.View.mViewFlags' on a null object reference
+        // So far I haven't been able to isolate the cause of this -- I think it might be related
+        // to removing the view immediately after an animation ends, but I haven't been able to
+        // reproduce it under controlled conditions.
```

Comment:
Posting to the view causes the runnable to occur at the end of the current queue of things it is doing. Basically, you're trying to mutate a view while it is likely to be in an immutable state. Doing this like this makes sense to me.